server.js

```diff
@@ -1,33 +1,98 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+
+import booksData from "./data/books.json";
+
+const ERROR_NOT_FOUND = { error: "No book results were found" };
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+const listEndpoints = require("express-list-endpoints");
 
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+// Add middlewares to enable cors and json body parsing - makes it easier for frontend to use the API. Corrs allows API's to say where requests can come from. To add extra security
+app.use(cors());
+app.use(bodyParser.json()); //This allows Express to read JSON in POST requests
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+app.get("/books", (req, res) => {
+  let page = req.query.page ?? 1;
```

Comment:
I'm impressed by seeing `??` - sometimes called nullish operator. Just to be sure you are aware of all features this operator offers and difference between `??` and `||`, check out https://www.javascripttutorial.net/es-next/javascript-nullish-coalescing-operator/